toonz/sources/toonz/penciltestpopup.cpp

```diff
@@ -1288,6 +1293,71 @@ void PencilTestPopup::onOnionCBToggled(bool on) {
 
 //-----------------------------------------------------------------------------
 
+void PencilTestPopup::onLoadImageButtonPressed() {
+  TCellSelection* cellSelection = dynamic_cast<TCellSelection*>(
+      TApp::instance()->getCurrentSelection()->getSelection());
+  if (cellSelection) {
+    int c0;
+    int r0;
+    TCellSelection::Range range = cellSelection->getSelectedCells();
+    if (range.isEmpty()) {
+      error(tr("No image selected.  Please select an image in the Xsheet."));
+      return;
+    }
+    c0 = range.m_c0;
+    r0 = range.m_r0;
+    TXshCell cell =
+        TApp::instance()->getCurrentXsheet()->getXsheet()->getCell(r0, c0);
+    if (cell.getSimpleLevel() == nullptr) {
+      error(tr("No image selected.  Please select an image in the Xsheet."));
+      return;
+    }
+    TXshSimpleLevel* level = cell.getSimpleLevel();
+    int type               = level->getType();
+    if (type != TXshLevelType::OVL_XSHLEVEL) {
+      error(tr("The selected image is not in a raster level."));
+      return;
+    }
+    TImageP origImage = cell.getImage(false);
+    TRasterImageP tempImage(origImage);
+    TRasterImage* image = (TRasterImage*)tempImage->cloneImage();
+
+    int m_lx          = image->getRaster()->getLx();
+    int m_ly          = image->getRaster()->getLy();
+    QString res       = m_resolutionCombo->currentText();
+    QStringList texts = res.split(' ');
+    if (m_lx != texts[0].toInt() || m_ly != texts[2].toInt()) {
+      error(
+          tr("The selected image size does not match the current camera "
+             "settings."));
+      return;
+    }
+    int m_bpp      = image->getRaster()->getPixelSize();
+    int totalBytes = m_lx * m_ly * m_bpp;
+    image->getRaster()->yMirror();
+
+    // lock raster to get data
+    image->getRaster()->lock();
+    void* buffin = image->getRaster()->getRawData();
+    assert(buffin);
+    void* buffer = malloc(totalBytes);
+    memcpy(buffer, buffin, totalBytes);
+
+    image->getRaster()->unlock();
+
+    QImage qi = QImage((uint8_t*)buffer, m_lx, m_ly, QImage::Format_ARGB32);
+    QImage qi2(qi.size(), QImage::Format_ARGB32);
+    qi2.fill(QColor(Qt::white).rgb());
+    QPainter painter(&qi2);
+    painter.drawImage(0, 0, qi);
```

Comment:
For now, if you are capturing images with the "Upside down" option, this button loads the previous image in the opposite direction.
How about to rotate the image if the "Upside down" option is ON, by adding lines like as follows?:

`    if (m_upsideDownCB->isChecked()){`
`      painter.translate(m_lx/2, m_ly/2);`
`      painter.rotate(180);`
`      painter.translate(-m_lx / 2, -m_ly / 2);`
`    }`